packages/aws-cdk-lib/core/lib/stack.ts

```diff
@@ -804,6 +804,21 @@ export class Stack extends Construct implements ITaggable {
     return value;
   }
 
+  /**
+   * Returns the list of resolved AZs that are available in the AWS environment
+   * (account/region) associated with this stack. If there are no resolved AZs,
+   * then an empty array will be returned.
+   *
+   * If no availability zones exist in the context, then they will be reported
+   * as missing and the CLI will resolve them by calling EC2 `DescribeAvailabilityZones`
+   * on the target environment.
+   */
+  public get resolvedAvailabilityZones(): string[] {
```

Comment:
After thinking about it some more I think this should just be a private method on the `Vpc` construct. It seems too specific of a use case to put on the `Stack` since most users should not need to care about the `dummy` values.

Reply:
Update made and changes pushed. I made this a private method as part of the Vpc construct. It's no longer a getter though. I decided the best way to structure this was as "resolveStackAvailabilityZones" which would then be passed the stack availability zones to be resolved.